tools/cibadmin.c

```diff
@@ -369,7 +424,45 @@ main(int argc, char **argv)
     } else if (admin_input_stdin) {
         source = "STDIN";
         input = stdin2xml();
+
+#if ENABLE_ACL
+    } else if (acl_eval_how != acl_eval_unused) {
+        username = uid2username(geteuid());
+        if (pcmk_acl_required(username)) {
+            if (force_flag == TRUE) {
+                fprintf(stderr, "The supplied command can provide skewed"
+                        " result since it is run under user that also gets"
+                        " guarded per ACLs on their own right."
+                        " Continuing since --force flag was provided.\n");
+                fflush(stderr);
+
+            } else {
+                fprintf(stderr, "The supplied command can provide skewed"
+                        " result since it is run under user that also gets"
+                        " guarded per ACLs in their own right."
+                        " To accept the risk of such a possible distortion"
+                        " (without even knowing it at this time), use"
+                        " the --force flag.\n");
+                fflush(stderr);
+                crm_exit(CRM_EX_UNSAFE);
+            }
+
+        }
+        free(username);
+        username = NULL;
+
+        if (cib_user == NULL) {
+            fprintf(stderr, "The supplied command requires -U user specified.\n");
+            fflush(stderr);
+            crm_exit(CRM_EX_USAGE);
+        }
+
+        /* we already stopped/warned ACL-controlled users about consequences */
+        acl_cred = cib_user;
+        cib_user = NULL;  /* XXX CRM_DAEMON_USER as it's IPC guarded anyway? */
+        /* XXX should likely differenciate per admin_input_* */
     }
+#endif
```

Comment:
The closing brace needs to be outside the guard

Reply:
Good catch